nwcsafpps_runner/tests/test_level1c_runner.py

```diff
@@ -328,14 +323,25 @@ def test_orbit_number_from_msg_viirs(self, cpu_count, config):
 
         input_msg = Message.decode(rawstr=TEST_INPUT_MESSAGE_VIIRS_MSG)
 
-        with patch('nwcsafpps_runner.l1c_processing.ThreadPool'):
-            with tempfile.NamedTemporaryFile() as myconfig_file:
-                l1c_proc = L1cProcessor(myconfig_file.name, 'viirs-l1c')
-                l1c_proc.run(input_msg)
+        with tempfile.NamedTemporaryFile() as myconfig_file:
+            l1c_proc = L1cProcessor(myconfig_file.name, 'viirs-l1c')
+            l1c_proc.run(input_msg)
 
         expected_orbit_number = TEST_VIIRS_MSG_DATA.get('orbit_number')
         self.assertEqual(l1c_proc.orbit_number, expected_orbit_number)
 
+    @patch('nwcsafpps_runner.config.load_config_from_file')
+    @patch.dict('nwcsafpps_runner.l1c_processing.LVL1C_PROCESSOR_MAPPING', {'viirs-l1c': my_fake_l1proc_function})
```

Comment:
Are you really sure you patch the correct place?

Reply:
The right place as in "is the function actually run by the test"? Yes, I am certain of that. If I change the time_limit_seconds: 1 to a higher value the test takes much longer as it is actually running the function sleep(65) and is interrupted only after time_limit_seconds. But there might be a better place to patch it though. I tried several worse places to patch before I came up with this one. For all other places I tried the my_fake_l1proc_function was never interrupted.